src/main/java/com/microsoft/azure/relay/HybridConnectionListener.java

```diff
@@ -36,6 +37,7 @@
 	private URI address;
 	private TrackingContext trackingContext;
 	private TokenProvider tokenProvider;
+	private Throwable injectedFault;
```

Comment:
>Throwable [](start = 9, length = 9)

Given the current code you could easily change this to just be UpgradeException. (would require one cast after the instanceof UgradeException in the `injectFault` method.

Reply:
Sounds like you want to keep it generic for future use. ok.

---
In reply to: [354522976](https://github.com/Azure/azure-relay-java/pull/58#discussion_r354522976) [](ancestors = 354522976)